File: src/api/account.ts
import { RequestFN } from "~/core/request-function";
import { decodeAccount } from "~/decoders/account";
import { type Account, type SessionHandle, TabLocation } from "~/models";

export const account = async (session: SessionHandle): Promise<Account> => {
  const request = new RequestFN(session, "PageInfosPerso", {
    _Signature_: {
      onglet: TabLocation.Account
      // TODO: Check if this is required in older version. On 2024, we apparently don't need it anymore.
      // ressource: {
      //   G: 4,
      //   N: input.userID
      // }
    }
  });

  const response = await request.send();
  return decodeAccount(response.data.donnees, session);
};

File: library/src/commonMain/kotlin/api/account.kt
package api

import core.RequestFN
import decoders.decodeAccount
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.*
import models.Account
import models.SessionHandle
import models.TabLocation

suspend fun account (session: SessionHandle): Account {
    val request = RequestFN(session.information, "PageInfosPerso", Json.encodeToString(buildJsonObject {
        putJsonObject("_Signature_") {
            put("onglet", TabLocation.Account.code)
        }
    }))

    val response = request.send()
    return decodeAccount(Json.parseToJsonElement(response.data).jsonObject["donnees"]!!.jsonObject, session)
}